### ISSUE TITLE
'kbox drush ard' paths don't seem to be correct

### ISSUE THREAD

**Issue by WidgetsBurritos** [kalabox/kalabox#924]
I'm still getting acquainted with kalabox and docker machine, so I'm hoping I'm not asking an overly obvious question, but I didn't find any info about it in the documentation. 

If I attempt to backup my data and files via drush, I would expect to be able to run the following command:
```
kbox drush ard
```

This definitely seems like it's doing something, and it acts like it works.   I get the following feedback: 
```
Database dump saved to /tmp/drush_tmp_1454364032_56afd5803f44a/pantheon.sql                                                                                                 [success]
Archive saved to /root/drush-backups/archive-dump/20160201220026/pantheon.20160201_220027.tar.gz                                                                            [ok]
/root/drush-backups/archive-dump/20160201220026/pantheon.20160201_220027.tar.gz
```

But when I look for those files, I can't find them. Where exactly are those paths supposed to reside?  I have checked both my local machine (Mac OS X - El Capitan) and my docker machine by logging in with the following command:

```
kbox terminal
```

I am currently running kalabox version 0.11.1.

I can always export the DB and file system directly from my machine if necessary, but just wanted to see what quicker options were available.

**Comment by pirog** [comment]
hey @WidgetsBurritos so the reason you aren't seeing anything is because `kbox drush ard` runs inside of a container and when the command is done that container gets destroyed, which also destroys your backup. 

There are, however, a few things you can do here. The following container directories are "persistent" and shared among all your containers

```
/code
/media
/certs
/sql
/backups
```

Your app root is also shared into the container at `/src`. So here are a few examples that might help you along

### Example 1

Exporting to a persistent directory and then accessing in another container

```
cd /path/to/app
kbox drush archive-dump default --destination=/backups/mysite.tar
kbox terminal
cd /backups
ls
# Should see backup
```

### Example 2

Export the archive to your app root. 

```
cd /path/to/app
drush archive-dump default --destination=/src/mysite.tar
ls
# should see archive exported 
```

**Comment by WidgetsBurritos** [comment]
Oh that completely makes sense. Do you think it would make sense to add something like this to the getting started guide? That seems like a somewhat common task that people may want to be able to do, and it may not be a bad idea to have it readily available.

**Comment by pirog** [comment]
yes! it would be great to add that into the wiki somewhere. is that something you have motivation to do?

**Comment by WidgetsBurritos** [comment]
Sure that will be fine.  It may take me a couple of days to get to it though.  Is that okay?

**Comment by pirog** [comment]
Absolutely! We appreciate all contributions.

**Comment by WidgetsBurritos** [comment]
I created a wiki page with the instructions for backing up: 
[https://github.com/kalabox/kalabox/wiki/Creating-backups-using-drush-within-Kalabox](url)

However, I was going to also include instructions on recovery and ran into an issue:

```
kbox drush archive-restore /src/backups/restore.tar
```

Will restore everything to `/src/code/code`, instead of inside `/src/code/`

Assuming this could be fixed by just specifying and alternate destination I run the following command:

```
Davids-MacBook-Pro:greenpawup davids$ kbox drush arr /src/backups/restore.tar --destination=/src/
info: COMPOSE =>  0=--project-name, 1=greenpawup, 2=--file, 3=/Users/davids/kalabox/apps/greenpawup/kalabox-compose.yml, 4=--file, 5=/Users/davids/kalabox/apps/greenpawup/kalabox-cli.yml, 6=--file, 7=/Users/davids/kalabox/apps/greenpawup/kalabox-cli.yml, 8=run, 9=-d, 10=--entrypoint, 11=tail, 12=terminus, 13=-f, 14=/dev/null, mode=false
Destination directory /src/ already exists.                          [error]
Unable to restore platform to /src/                                  [error]
```

So then I try adding the --overwrite flag to it, thinking it would save over the non existent files. 

```
Davids-MacBook-Pro:greenpawup davids$ kbox drush arr /src/backups/restore.tar --destination=/src/ --overwrite
info: COMPOSE =>  0=--project-name, 1=greenpawup, 2=--file, 3=/Users/davids/kalabox/apps/greenpawup/kalabox-compose.yml, 4=--file, 5=/Users/davids/kalabox/apps/greenpawup/kalabox-cli.yml, 6=--file, 7=/Users/davids/kalabox/apps/greenpawup/kalabox-cli.yml, 8=run, 9=-d, 10=--entrypoint, 11=tail, 12=terminus, 13=-f, 14=/dev/null, mode=false
rmdir(/src/): Device or resource busy filesystem.inc:149             [warning]
Destination directory /src/ already exists.                          [error]
Unable to move /tmp/drush_tmp_1455563732_56c223d492678/code to /src/.[error]
Unable to restore platform to /src/                                  [error]
```

And as it turns it, the entire /src/ directory got wiped out in the process.  YAY!

I'm guessing it has something to do with /src/ being a persistent directory inside kalabox.  

Wiping out the entire app file is definitely not desired behavior. 

I'm trying to work out a solution to the problem, but if you have any pointers I'd appreciate it.

**Comment by WidgetsBurritos** [comment]
So I came up with a temporary solution, but I'm not quite so sure it's optimal: 

```
kbox drush archive-restore /src/backups/restore.tar --destination=/backups/restore-tmp --overwrite
kbox terminal
rsync -az --del /backups/restore-tmp/ /code/
rm -Rf /backups/restore-tmp/
```

Basically the restore command handles the DB import and then imports the files to `/backups/restore-tmp/` so we can access them via the docker terminal, after which we use rsync to sync /backups/restore-tmp/ and /code/.  

Although this brings up another question, what is the difference between /code/ and /src/code/?

Basically I can see that /src/code is obviously what maps to my local machine and /code/ is where the site is web server reading from, and /src/code is missing sites/default/files? 
They seemed to be linked (although there is a slightly delayed linkage between the two).   

I just want to make sure by rsync'ing to /code/ all of the necessary files get copied over to /src/code/ as well. 

This solution technically works, but is definitely less than ideal as it would be nice to have a way to use `drush archive-restore` to handle both the code and the db.

**Comment by pirog** [comment]
Using a drush alias here seems like it would be the obvious way to go. If not you'll probably want to restore the files to your webroot which is `/code/`. 

@WidgetsBurritos whats the actual use case here? Why import a drush archive vs pulling what you need directly from pantheon?

**Comment by WidgetsBurritos** [comment]
Well in my case, it's just about being redundant. 

Suppose I pull from pantheon on Monday. I make changes Tuesday, Wednesday and Thursday, and push back on Friday. 

If for some reason if my computer crashes on Friday before I've had the chance to push back up to pantheon and all those changes are lost, I'd rather not lose three days worth of work. 

But I can set my machine to automatically create daily dumps of my local dev process, so that way come Friday, if my machine is hosed, I have all the readily available on another machine on my network.

**Comment by pirog** [comment]
Have you tried continuous integration or scheduling regular backups on pantheon? Also, exporting your database configuration to code using features or config management is usually another good first line of defense for this sort of thing.

**Comment by WidgetsBurritos** [comment]
The scheduling backups on pantheon only helps me when those changes have actually been pushed up to pantheon, which I don't do every day. I probably should do that more, but it's been more an issue of time.

I definitely agree on using features and will once all is said and done, but the project in question is a project that hasn't been launched yet and is still under development, so right now my local dev is the only version that really matters.  So I'm basically periodically just replacing everything on pantheon in case I need to show somebody something.
 
Right now I just have a local bash script on my system scheduled to run at 10pm every night and backs up my local dev and copies it to another machine on my network, which works even if my internet connection crashes. 

It's not an ideal workflow by any means, but it works for me right now, until I have time to implement a better workflow process.
